ios/FluentUI/Badge Field/BadgeField.swift

```diff
@@ -316,6 +327,15 @@ open class BadgeField: UIView {
         return CGSize(width: UIView.noIntrinsicMetric, height: contentHeight(forBoundingWidth: frame.width))
     }
 
+    open override func willMove(toWindow newWindow: UIWindow?) {
+        super.willMove(toWindow: newWindow)
+        guard let newWindow else {
+            return
+        }
+        tokenSet.update(newWindow.fluentTheme)
+        updateAppearance()
```

Comment:
I think you shouldn't need the `updateAppearance()` here, since setting the theme through `tokenSet.update` should trigger what you added with `registerOnUpdate` if the theme changes.